File: frontend/src/types/sale.ts
export interface CheckoutInput {
  items: Array<{ product_id: string; quantity: number }>
  amount_tendered: string
}

export interface SaleItem {
  id: string
  product_id: string
  product_name: string
  sku: string
  unit_price: string
  quantity: number
  line_total: string
}

export interface Sale {
  id: string
  receipt_number: string
  subtotal: string
  total: string
  amount_tendered: string
  change_due: string
  payment_method: 'cash'
  created_at: string
  items: SaleItem[]
}

export interface SaleSummary {
  id: string
  receipt_number: string
  created_at: string
  payment_method: string
  total: string
  item_count: number
}

export interface SalesPage {
  items: SaleSummary[]
  page: number
  page_size: number
  total_items: number
  total_pages: number
}

File: backend/app/schemas/sale.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class CheckoutItem(BaseModel):
    product_id: uuid.UUID
    quantity: int = Field(gt=0)


class CheckoutCreate(BaseModel):
    items: list[CheckoutItem] = Field(min_length=1)
    amount_tendered: Decimal = Field(ge=0, max_digits=12, decimal_places=2)


class SaleItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    product_id: uuid.UUID
    product_name: str
    sku: str
    unit_price: Decimal
    quantity: int
    line_total: Decimal


class SaleRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    receipt_number: str
    subtotal: Decimal
    total: Decimal
    amount_tendered: Decimal
    change_due: Decimal
    payment_method: str
    created_at: datetime
    items: list[SaleItemRead]
